Mage.Sets/src/mage/cards/l/LilianaWakerOfTheDead.java

```diff
@@ -0,0 +1,116 @@
+        this.addAbility(new LoyaltyAbility(new LilianaWakerOfTheDeadDiscardEffect(), 1));
+
+        // −3: Target creature gets -X/-X until end of turn, where X is the number of cards in your graveyard.
+        Ability ability = new LoyaltyAbility(new BoostTargetEffect(
+                xValue, xValue, Duration.EndOfTurn, true
+        ).setText("target creature gets -X/-X until end of turn, where X is the number of cards in your graveyard"), -3);
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+
+        // −7: You get an emblem with "At the beginning of combat on your turn, put target creature card from a graveyard onto the battlefield under your control. It gains haste."
+        this.addAbility(new LoyaltyAbility(new GetEmblemEffect(new LilianaWakerOfTheDeadEmblem()), -7));
+    }
+
+    private LilianaWakerOfTheDead(final LilianaWakerOfTheDead card) {
+        super(card);
+    }
+
+    @Override
+    public LilianaWakerOfTheDead copy() {
+        return new LilianaWakerOfTheDead(this);
+    }
+}
+
+class LilianaWakerOfTheDeadDiscardEffect extends OneShotEffect {
+
+    LilianaWakerOfTheDeadDiscardEffect() {
+        super(Outcome.Discard);
+        staticText = "Each player discards a card. Each opponent who can't loses 3 life";
+    }
+
+    private LilianaWakerOfTheDeadDiscardEffect(LilianaWakerOfTheDeadDiscardEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LilianaWakerOfTheDeadDiscardEffect copy() {
+        return new LilianaWakerOfTheDeadDiscardEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Map<UUID, Cards> cardsToDiscard = new HashMap<>();
+        if (controller == null) {
+            return true;
+        }
+        // choose cards to discard
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) {
+                continue;
+            }
+            int numberOfCardsToDiscard = Math.min(1, player.getHand().size());
+            Cards cards = new CardsImpl();
+            Target target = new TargetDiscard(numberOfCardsToDiscard, numberOfCardsToDiscard, StaticFilters.FILTER_CARD, playerId);
+            player.chooseTarget(outcome, target, source, game);
+            cards.addAll(target.getTargets());
+            cardsToDiscard.put(playerId, cards);
+        }
+        // discard all choosen cards
+        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) {
+                continue;
+            }
+            int amountDiscarded = player.discard(cardsToDiscard.get(playerId), source, game).size();
+            if (!playerId.equals(controller.getId()) && amountDiscarded == 0) {
```

Comment:
This should use `Player::hasOpponent` since we want it to be compatible with the 2HG mode that we'll never get around to implementing.

Reply:
ah, gotcha, fixed!